### PR TITLE
reset time after each run and panic on leader errors

### PR DESCRIPTION
**Pull request by chongr**

### CODE REVIEW

File: firehose.go [hunk 1]
```diff
@@ -385,7 +393,10 @@ outer:
 	}
 	if leader {
 		// 2. if leader call RunLeader
-		fp.RunLeader(ctx)
+		err := fp.RunLeader(ctx)
+		if err != nil && err != context.Canceled && err != context.DeadlineExceeded {
+			panic(err)
```
Comment (maroux) [inline firehose.go]: Don't love the panic here - can we just bubble this up instead?

File: firehose.go [hunk 2]
```diff
@@ -286,6 +293,7 @@ func (fp *Firehose) RunLeader(ctx context.Context) {
 			}
 			wg.Wait()
 			// restart scrape interval and run leader again
```
Comment (maroux) [inline firehose.go:295]: comment needs to move one line below?

File: firehose.go (hunk 1 above)
Comment (chongr) [reply firehose.go]: ah yeah this is the top level the follower also panics, should I return the error from this function?

Comment (maroux) [reply firehose.go]: oh hm yeah I think updating function signature is better.

File: firehose.go (hunk 2 above)
Comment (chongr) [reply firehose.go:295]: I sort of considered both resetting current time and the timer channel as "restarting the scrape interval"

Comment (chongr) [reply firehose.go:295]: because currentttime needs to be reset too otherwise the file is just overwritten

**Review by maroux — DISMISSED** [review]

File: firehose.go (hunk 2 above)
Comment (maroux) [reply firehose.go:295]: makes sense.

**Review by maroux — APPROVED** [review]

**Pull request merged**